### PR TITLE
chain: refactor `process_partial_encoded_chunk_request` method

### PR DESCRIPTION
**Pull request by mina86**
Extract two more methods from `process_partial_encoded_chunk_request`
which correspond to `if` bodies that used to be in it.  This makes each
function shorter and thus easier to read especially as in the future
more branches will be added to the method.

Furthermore, move sending of the message to the method changing
`maybe_send_partial_encoded_chunk_response` into method which prepares
the response only.

This is a pure refactoring with no changes to the behaviour.

Issue: https://github.com/near/nearcore/issues/6242

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -998,78 +998,106 @@ impl ShardsManager {
...
-                .map(|shard_id| present_receipts.get(shard_id).map(|x| *x).cloned());
-
-            // Pass iterators to function, same as cache case.
-            return self.maybe_send_partial_encoded_chunk_response(
-                request.chunk_hash,
-                route_back,
-                parts_iter,
-                receipts_iter,
-            );
+            Self::prepare_partial_encoded_chunk_response_from_partial(request, partial_chunk)
```
Comment (mm-near) [inline chain/chunks/src/lib.rs:1004]: nit: I'd call it "_from_storage" rather than "from_partial"

File: chain/chunks/src/lib.rs [hunk 2]
```diff
@@ -998,78 +998,106 @@ impl ShardsManager {
...
+        } else {
+            None
         };
+
+        if let Some(response) = response {
+            self.peer_manager_adapter.do_send(PeerManagerMessageRequest::NetworkRequests(
+                NetworkRequests::PartialEncodedChunkResponse { route_back, response },
+            ))
+        }
```
Comment (mm-near) [inline chain/chunks/src/lib.rs:1013]: in separate PR:

maybe we should have some stats - on how many times we don't have the response (as normally this should not happen).

File: chain/chunks/src/lib.rs [hunk 3]
```diff
@@ -998,78 +998,106 @@ impl ShardsManager {
...
-    ) where
+    ) -> Option<PartialEncodedChunkResponseMsg>
+    where
         A: Iterator<Item = Option<PartialEncodedChunkPart>>,
         B: Iterator<Item = Option<ReceiptProof>>,
     {
         let maybe_known_parts: Option<Vec<_>> = parts_iter.collect();
         let parts = match maybe_known_parts {
             None => {
                 debug!(target:"chunks", "Not responding, some parts are missing");
```
Comment (mm-near) [inline chain/chunks/src/lib.rs:1099]: maybe we should also add 'chunk_hash' to those debug logs

**Review by mm-near — APPROVED** [review]

File: chain/chunks/src/lib.rs (hunk 1 above)
Comment (mina86) [reply chain/chunks/src/lib.rs:1004]: Problem with that name is that I’m working on `_from_chunk` method which also reads data from storage.

**Pull request merged**